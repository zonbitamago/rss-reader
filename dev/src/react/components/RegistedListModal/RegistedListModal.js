import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./RegistedListModal.css";
import Dialog, {
  DialogActions,
  DialogContent,
  DialogTitle
} from "material-ui/Dialog";
import Input from "../Input/Input";
import Button from "../Button/Button";

class RegistedListModal extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="RegistedListModal">
        <Dialog
          open={this.props.open}
          onClose={this.props.handleClose}
          aria-labelledby="form-dialog-title">
          <DialogTitle id="form-dialog-title">RegistedListModal</DialogTitle>
          <DialogContent>
            <Input name="Name" />
            <Input name="URL" />
          </DialogContent>
          <DialogActions>
            <Button type="yes" handleClick={this.props.handleClose} />
            <Button type="no" handleClick={this.props.handleClose} />
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

RegistedListModal.propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func
};

RegistedListModal.defaultProps = {};

export default RegistedListModal;
