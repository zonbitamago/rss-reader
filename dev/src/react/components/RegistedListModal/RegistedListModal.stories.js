import React from "react";
import { storiesOf } from "@storybook/react";

import RegistedListModal from "./RegistedListModal";

storiesOf("Organisms/RegistedListModal", module).add(
  "RegistedListModal",
  () => (
    <div>
      <RegistedListModal open={true} handleClose={() => {}} />
    </div>
  )
);
