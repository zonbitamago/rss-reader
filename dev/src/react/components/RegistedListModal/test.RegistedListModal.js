import React from 'react'
import { shallow } from 'enzyme'

import RegistedListModal from './RegistedListModal'

describe('RegistedListModal', () => {
  let component, props

  beforeEach(() => {
    props = {}
    component = shallow(<RegistedListModal {...props} />)
  })

  it('should', () => {
    expect(component).toMatchSnapshot()
  })
})