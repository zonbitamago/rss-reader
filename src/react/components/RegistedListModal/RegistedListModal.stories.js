// jest.mock("rc-util/lib/Portal");
import React from "react";
import { storiesOf } from "@storybook/react";
import RegistedListModal from "./RegistedListModal";
import { electronUtil } from "../../../../__mocks__/electronUtilSettings";
import { store } from "../../../../__mocks__/storeSettings";

storiesOf("Organisms/RegistedListModal", module).add("open", () => (
  <div>
    <RegistedListModal
      open={true}
      handleClose={() => {
        console.log("test");
      }}
      store={store}
      electronUtil={electronUtil}
    />
  </div>
));
