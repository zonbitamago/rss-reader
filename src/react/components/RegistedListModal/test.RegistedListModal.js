import React from "react";
import { mount } from "enzyme";

import RegistedListModal from "./RegistedListModal";
import { electronUtil } from "../../../../__mocks__/electronUtilSettings";
import { store } from "../../../../__mocks__/storeSettings";

jest.mock("../../../../__mocks__/storeSettings");

describe("RegistedListModal", () => {
  let wrapper, props, mockfn;

  beforeEach(() => {
    mockfn = jest.fn();
    store.FeedListStore.url = "";
    store.FeedListStore.name = "";
    store.FeedListStore.feedList = [
      {
        name: "name",
        url: "url"
      }
    ];
    store.FeedListStore.setFeedList.mockRestore();
    store.ItemStore.setTimer.mockRestore();
    props = {
      open: true,
      handleClose: mockfn,
      store: store,
      electronUtil: electronUtil
    };
    wrapper = mount(<RegistedListModal {...props} />);
  });

  it("yesボタンクリック", () => {
    store.FeedListStore.setFeedList.mockImplementation(() => {
      return new Promise(resolve => {
        resolve(true);
      });
    });
    expect(mockfn.mock.calls.length).toBe(0);
    expect(store.FeedListStore.setFeedList.mock.calls.length).toBe(0);
    // expect(store.ItemStore.setTimer.mock.calls.length).toBe(0);

    wrapper.find(".Button > #yes").simulate("click");

    expect(mockfn.mock.calls.length).toBe(0);
    expect(store.FeedListStore.setFeedList.mock.calls.length).toBe(1);
    // expect(store.ItemStore.setTimer.mock.calls.length).toBe(1);

    store.FeedListStore.setFeedList.mockImplementation(() => {
      return new Promise(resolve => {
        resolve(false);
      });
    });

    wrapper.find(".Button > #yes").simulate("click");
    expect(mockfn.mock.calls.length).toBe(0);
    expect(store.FeedListStore.setFeedList.mock.calls.length).toBe(2);
  });

  it("noボタンクリック", () => {
    expect(mockfn.mock.calls.length).toBe(0);
    wrapper.find(".Button > #no").simulate("click");
    expect(mockfn.mock.calls.length).toBe(1);
  });

  it("URL Input Change", () => {
    expect(store.FeedListStore.url).toBe("");
    wrapper
      .find(".RegistedListModal input#URL")
      .simulate("change", { target: { value: "test" } });
    expect(store.FeedListStore.url).toBe("test");
  });

  it("Name Input Change", () => {
    expect(store.FeedListStore.name).toBe("");
    wrapper
      .find(".RegistedListModal input#Name")
      .simulate("change", { target: { value: "test" } });
    expect(store.FeedListStore.name).toBe("test");
  });
});
