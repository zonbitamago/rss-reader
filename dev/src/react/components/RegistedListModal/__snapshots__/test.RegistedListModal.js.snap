// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistedListModal should 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegistedListModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        aria-labelledby="form-dialog-title"
        onClose={undefined}
        open={undefined}
      >
        <WithStyles(DialogTitle)
          id="form-dialog-title"
        >
          RegistedListModal
        </WithStyles(DialogTitle)>
        <WithStyles(DialogContent)>
          <Input
            name="Name"
          />
          <Input
            name="URL"
          />
        </WithStyles(DialogContent)>
        <WithStyles(DialogActions)>
          <Button
            type="yes"
          />
          <Button
            type="no"
          />
        </WithStyles(DialogActions)>
      </WithStyles(Dialog)>,
      "className": "RegistedListModal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-labelledby": "form-dialog-title",
        "children": Array [
          <WithStyles(DialogTitle)
            id="form-dialog-title"
          >
            RegistedListModal
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <Input
              name="Name"
            />
            <Input
              name="URL"
            />
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <Button
              type="yes"
            />
            <Button
              type="no"
            />
          </WithStyles(DialogActions)>,
        ],
        "onClose": undefined,
        "open": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "RegistedListModal",
            "id": "form-dialog-title",
          },
          "ref": null,
          "rendered": "RegistedListModal",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Input
                name="Name"
              />,
              <Input
                name="URL"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "Name",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "URL",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Button
                type="yes"
              />,
              <Button
                type="no"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "type": "yes",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "type": "no",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          aria-labelledby="form-dialog-title"
          onClose={undefined}
          open={undefined}
        >
          <WithStyles(DialogTitle)
            id="form-dialog-title"
          >
            RegistedListModal
          </WithStyles(DialogTitle)>
          <WithStyles(DialogContent)>
            <Input
              name="Name"
            />
            <Input
              name="URL"
            />
          </WithStyles(DialogContent)>
          <WithStyles(DialogActions)>
            <Button
              type="yes"
            />
            <Button
              type="no"
            />
          </WithStyles(DialogActions)>
        </WithStyles(Dialog)>,
        "className": "RegistedListModal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-labelledby": "form-dialog-title",
          "children": Array [
            <WithStyles(DialogTitle)
              id="form-dialog-title"
            >
              RegistedListModal
            </WithStyles(DialogTitle)>,
            <WithStyles(DialogContent)>
              <Input
                name="Name"
              />
              <Input
                name="URL"
              />
            </WithStyles(DialogContent)>,
            <WithStyles(DialogActions)>
              <Button
                type="yes"
              />
              <Button
                type="no"
              />
            </WithStyles(DialogActions)>,
          ],
          "onClose": undefined,
          "open": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "RegistedListModal",
              "id": "form-dialog-title",
            },
            "ref": null,
            "rendered": "RegistedListModal",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Input
                  name="Name"
                />,
                <Input
                  name="URL"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "Name",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "URL",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Button
                  type="yes"
                />,
                <Button
                  type="no"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "type": "yes",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "type": "no",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
